biz.ganttproject.core/src/main/java/biz/ganttproject/core/chart/scene/gantt/TableSceneBuilder.kt

```diff
@@ -19,40 +19,104 @@ along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
 package biz.ganttproject.core.chart.scene.gantt
 
 import biz.ganttproject.core.chart.canvas.Canvas
+import biz.ganttproject.core.chart.canvas.TextMetrics
+import java.lang.Integer.max
 
 class TableSceneBuilder(private val config: Config) {
-  fun build(rows: List<Row>, canvas: Canvas = Canvas()): Canvas {
-    val dimensions = calculateDimensions(rows)
+  private lateinit var table: Table
+  private lateinit var canvas: Canvas
+  private lateinit var colsWidth: Map<Table.Column, Int>
+  private lateinit var dimensions: Dimension
```

Comment:
Didn't like passing this 4 parameters to the paint methods, so ended up turning them into fields

Reply:
This means that we need to be careful with using the instance of this builder and do not allow for concurrent calls to `build`. In particular, it will make sense to construct the instance every time in `TaskTableSceneBuilder::build` instead of using a single instance as it is done now.

Reply:
Then maybe move `table` to constructor arguments and create a `TableSceneBuilder`  for each `build` call?

Reply:
Sounds reasonable.